packages/react-core-notifications/src/InAppMessaging/hooks/useMessage/__tests__/useMessage.spec.ts

```diff
@@ -31,6 +27,10 @@ const infoSpy = jest.spyOn(Logger.prototype, 'info');
 
 const mockUseInAppMessaging = useInAppMessaging as jest.Mock;
 const mockClearMessage = jest.fn();
+jest.mock('aws-amplify/in-app-messaging', () => ({
+  ...jest.requireActual('aws-amplify/in-app-messaging'),
+  notifyMessageInteraction: jest.fn(),
+}));
```

Comment:
Prefer spies over mocks, spies retain type information and can catch breaking changes that a mock would hide

Reply:
I couldn't figure out how to do that with the functional apis because spyOn expects an object and a method